exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -118,13 +133,34 @@ public boolean apply(@Nullable FileStatus status) {
       }
     }));
 
-    return create(nonDirectories, null, selectionRoot);
+    final FileSelection fileSel = create(nonDirectories, null, selectionRoot);
```

Comment:
Does it make sense to pass in the list of Files in addition to the list of FileStatus? I thought the implicit assumption for FileSelection is list of files goes first before FileStatus.